bindings/native/examples/main.rs.bak

```diff
@@ -0,0 +1,77 @@
+use std::{collections::HashMap, borrow::Cow};
+
+use hotg_rune_core::Shape;
+
+struct Tensor {
+    shape: Shape<'static>,
+    buffer: Vec<u8>,
+}
+
+fn main() {
+    let rune: Vec<u8> = download_rune();
+
+    let mut runtime = Runtime::load(
+        &rune,
+        HostFunctions {
+            log: default_log,
+            preprocess_model: None,
+        },
+    )
+    .unwrap();
+
+    let capabilities: HashMap<CapabilityId, CapabilityInfo> =
+        runtime.capabilities();
+
+    // TODO: give users a way to read the RuneGraph and normal "rune inspect"
+    // metadata
+
+    let inputs: HashMap<CapabilityId, Tensor> = HashMap::new();
```

Comment:
Looks good. But let the runtime allocate inputs/outputs and then give access to those buffers?